examples/preprocessing/plot_rereference_eeg.py

```diff
@@ -44,18 +44,22 @@
 raw.set_eeg_reference([])
 evoked_no_ref = mne.Epochs(raw, **epochs_params).average()
 
-evoked_no_ref.plot(axes=ax1, titles=dict(eeg='EEG Original reference'))
+evoked_no_ref.plot(axes=ax1, titles=dict(eeg='EEG Original reference'),
+                   show=False)
 
 # Average reference. This is normally added by default, but can also be added
 # explicitly.
-raw.set_eeg_reference('average', projection=True)
+raw.set_eeg_reference('average', projection=False)
```

Comment:
For now we should continue to prefer `projection=True` as it's the only one that works with inverse solvers

Reply:
But in this example `projection=False` makes more sense because we are not following up with any inverse methods. Also, `projection=False` by default in 0.16 (at least that's what we said).

Reply:
Can you update the comment to mention this inverse business explicitly?

Reply:
Sure, but is this necessary in this example? This started out as an innocent demo of 3 different EEG references, and adding a statement about how awkward the function really behaves might confuse people more than help 😄 .

Reply:
Oh, or did you mean in the docstring of `set_eeg_reference`?

Reply:
I meant here -- I think everywhere else we use `sample` data we usually use `projection=True` (implicitly or explicitly), because there are many examples where it does get used for inverse modeling. Having `projection=False` here breaks that pattern, thus why it seems nice to give a bit more explanation to people.

Reply:
Why not use the `apply_proj` function? Thus, `projection=True` would work.

Reply:
True if you know the whole picture and aren't just looking at this specific example. I guess I could leave `projection=True` because it gets applied automatically when epoching anyway.

Reply:
> I guess I could leave projection=True because it gets applied automatically when epoching anyway.

Sure. Although in this case we should probably add `proj=True` in the `Epochs` constructor (even though it's the default) if it's not already there to make it explicit that it will happen. Either way -- `projection=False` with narrative update or `True` with `Epochs` update if necessary -- is fine by me

Reply:
I am with @larsoner about being more upfront about why some things are needed (at the cost of verbosity).